Misc/aflytning-keyboard-warrior/index.md

```diff
@@ -0,0 +1,209 @@
+
+Manually manipulating the data  down to the fields that actually change between lines, I was left with this format:
+```
+020000
+020e00
+020000
+000400
+000800
+001500
+001508
+000800
+002c00
+020000
+...
+```
+
+The first byte is a modifier key. For example pressing shift, produces `020000`, whilst pressing another key with shift held down, produces `020e00`.
+
+The second byte is the character key being pressed, the bytes matched almost all tables I found online (eg. `04 == a`), so I'm guessing its a standard for american keyboards.
+
+The third character seems to be a repetition-marker. Every time its greater than 0, the key seems to be held down, and will be duplicated if transferred to the output.
+
+Based on that, I wrote this very [rudimentary Python script](scripts/decode_keyboard_v1.py):
+
+```python
+import os
+
+presses = []
+with open(os.path.relpath("capdata.txt"), "r") as f:
+    for line in f:
+        presses.append(line[0:-1])
+
+normalKeys = {"04":"a", "05":"b", "06":"c", "07":"d", "08":"e", "09":"f", "0a":"g", "0b":"h", "0c":"i", "0d":"j", "0e":"k", "0f":"l", "10":"m", "11":"n", "12":"o", "13":"p", "14":"q", "15":"r", "16":"s", "17":"t", "18":"u", "19":"v", "1a":"w", "1b":"x", "1c":"y", "1d":"z","1e":"1", "1f":"2", "20":"3", "21":"4", "22":"5", "23":"6","24":"7","25":"8","26":"9","27":"0","28":"<RET>","29":"<ESC>","2a":"<DEL>", "2b":"\t","2c":"<SPACE>","2d":"-","2e":"=","2f":"[","30":"]","31":"\\","32":"<NON>","33":";","34":"'","35":"<GA>","36":",","37":".","38":"/","39":"<CAP>","3a":"<F1>","3b":"<F2>", "3c":"<F3>","3d":"<F4>","3e":"<F5>","3f":"<F6>","40":"<F7>","41":"<F8>","42":"<F9>","43":"<F10>","44":"<F11>","45":"<F12>"}
+shiftKeys = {"04":"A", "05":"B", "06":"C", "07":"D", "08":"E", "09":"F", "0a":"G", "0b":"H", "0c":"I", "0d":"J", "0e":"K", "0f":"L", "10":"M", "11":"N", "12":"O", "13":"P", "14":"Q", "15":"R", "16":"S", "17":"T", "18":"U", "19":"V", "1a":"W", "1b":"X", "1c":"Y", "1d":"Z","1e":"!", "1f":"@", "20":"#", "21":"$", "22":"%", "23":"^","24":"&","25":"*","26":"(","27":")","28":"<RET>","29":"<ESC>","2a":"<DEL>", "2b":"\t","2c":"<SPACE>","2d":"_","2e":"+","2f":"{","30":"}","31":"|","32":"<NON>","33":":","34":"\"","35":"<GA>","36":"<","37":">","38":"?","39":"<CAP>","3a":"<F1>","3b":"<F2>", "3c":"<F3>","3d":"<F4>","3e":"<F5>","3f":"<F6>","40":"<F7>","41":"<F8>","42":"<F9>","43":"<F10>","44":"<F11>","45":"<F12>"}
+
+def main():
+    result = ""
+    for press in presses:
+        bytes = [press[i:i+2] for i in range(0, len(press), 2)]
+
+        if bytes[1] == "00" or bytes[2] != "00":
+            continue
+    
+        if bytes[0] == "02":
+            result += shiftKeys[bytes[1]]
+        else:
+            result += normalKeys[bytes[1]]
+
+    print("[+] Found : %s" % (result))
+
+if __name__ == "__main__":
+    main()
+```
+
+Which produces this output:
+```
```

Comment:
Same suggestion
```suggestion
```text
```

This is absolutely hilarious, because of the edits! :D

Reply:
Maybe consider adding a note to read about the edits? I almost missed them on the first read! :D